internal/modelcontroller/patch_test.go

```diff
@@ -0,0 +1,117 @@
+package modelcontroller
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/substratusai/kubeai/internal/config"
+	"github.com/substratusai/kubeai/internal/k8sutils"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+)
+
+func Test_applyJSONPatchToPod(t *testing.T) {
+	t.Parallel()
+
+	emptyPod := &corev1.Pod{}
+	emptyPodHash := k8sutils.PodHash(emptyPod.Spec)
+
```

Comment:
Not sure if this is overkill, but thought I'd reuse the pod hash util to have a better guarantee that the pod has not been modified.